### ISSUE TITLE
Small problem when trying to do "gem install gtk2" - can not find the header file rbglibdeprecated.h

### ISSUE THREAD

**Issue by shevegen** [ruby-gnome2/ruby-gnome2#1123]
Hello.

Small problem I get when I do:

```console
% gem install gtk2
Building native extensions. This could take a while...  
Successfully installed pango-3.2.1  
Building native extensions. This could take a while...  
ERROR:  Error installing gtk2:  


  current directory: /Programs/Ruby/2.5.0/lib/ruby/gems/2.5.0/gems/gtk2-3.2.1/ext/gtk2  
/Programs/Ruby/2.5.0/bin/ruby -r ./siteconf20180106-20245-10y6z.rb extconf.rb 
checking for --enable-debug-build option... no  
checking for -Wall option to compiler... yes  
checking for -Waggregate-return option to compiler... yes 
checking for -Wcast-align option to compiler... yes 
checking for -Wextra option to compiler... yes  
checking for -Wformat=2 option to compiler... yes 
checking for -Winit-self option to compiler... yes  
checking for -Wlarger-than-65500 option to compiler... yes  
checking for -Wmissing-declarations option to compiler... yes 
checking for -Wmissing-format-attribute option to compiler... yes 
checking for -Wmissing-include-dirs option to compiler... yes 
checking for -Wmissing-noreturn option to compiler... yes 
checking for -Wmissing-prototypes option to compiler... yes 
checking for -Wnested-externs option to compiler... yes 
checking for -Wold-style-definition option to compiler... yes 
checking for -Wpacked option to compiler... yes 
checking for -Wp,-D_FORTIFY_SOURCE=2 option to compiler... yes  
checking for -Wpointer-arith option to compiler... yes  
checking for -Wswitch-default option to compiler... yes 
checking for -Wswitch-enum option to compiler... yes  
checking for -Wundef option to compiler... yes  
checking for -Wout-of-line-declaration option to compiler... no 
checking for -Wunsafe-loop-optimizations option to compiler... yes  
checking for -Wwrite-strings option to compiler... yes  
checking for rb_errinfo()... yes  
checking for cairo... yes 
checking for rb_cairo.h... yes  
checking for Windows... no  
checking for gtk+-2.0 version (>= 2.10.0)... yes  
checking for st.h... yes  
checking for ruby/st.h... yes 
checking for target... x11  
checking for gtk_plug_get_type() in gtk/gtk.h... yes  
checking for gtk_socket_get_type() in gtk/gtk.h... yes  
checking for pango_render_part_get_type() in gtk/gtk.h... yes 
checking for gtk/gtkfilesystem.h... no  
checking for X11/Xlib.h... yes  
checking for -lX11... yes 
checking for XReadBitmapFileData() in X11/Xlib.h... yes 
checking for XGetErrorText() in X11/Xlib.h... yes 
checking for gtk+-unix-print-2.0... yes 
creating ruby-gtk2.pc 
creating Makefile 

current directory: /Programs/Ruby/2.5.0/lib/ruby/gems/2.5.0/gems/gtk2-3.2.1/ext/gtk2  
make "DESTDIR=" clean 

current directory: /Programs/Ruby/2.5.0/lib/ruby/gems/2.5.0/gems/gtk2-3.2.1/ext/gtk2  
make "DESTDIR=" 
compiling rbgtkbox.c  
rbgtkbox.c: In function 'box_pack_start_or_end':  
rbgtkbox.c:36:16: warning: variable 'widget' set but not used [-Wunused-but-set-variable] 
 GtkWidget *widget, *child; 
  ^~~~~~  
rbgtkbox.c:41:17: warning: this statement may fall through [-Wimplicit-fallthrough=]  
 padding = NUM2INT(arg3); 
rbgtkbox.c:42:7: note: here 
 case 3:  
 ^~~~ 
rbgtkbox.c:43:14: warning: this statement may fall through [-Wimplicit-fallthrough=]  
 fill = RVAL2CBOOL(arg2); 
rbgtkbox.c:44:7: note: here 
 case 2:  
 ^~~~ 
rbgtkbox.c:45:16: warning: this statement may fall through [-Wimplicit-fallthrough=]  
 expand = RVAL2CBOOL(arg1); 
rbgtkbox.c:46:7: note: here 
 default: 
 ^~~~~~~  
compiling rbgtkaction.c 
compiling rbgtkfilesel.c  
compiling rbgtktextchild.c  
compiling rbgtkallocation.c 
compiling rbgtktoolbutton.c 
compiling rbgtkviewport.c 
compiling rbgtkvseparator.c 
compiling rbgtkselectiondata.c  
compiling rbgtklabel.c  
compiling rbgtkwindow.c 
rbgtkwindow.c: In function 'Init_gtk_window': 
rbgtkwindow.c:585:13: warning: 'rb_data_object_alloc' is deprecated: by rb_data_object_wrap [-Wdeprecated-declarations] 
 rb_data_object_alloc(rb_cData, NULL, mark_toplevels, NULL);  
 ^~~~~~~~~~~~~~~~~~~~ 
In file included from /Programs/Ruby/2.5.0/include/ruby-2.5.0/ruby.h:33:0,  
 from /Programs/Ruby/2.5.0/lib/ruby/gems/2.5.0/gems/glib2-3.2.1/ext/glib2/rbgobject.h:27, 
 from rbgtk.h:28, 
 from global.h:27,  
 from rbgtkwindow.c:24: 
/Programs/Ruby/2.5.0/include/ruby-2.5.0/ruby/ruby.h:1385:1: note: declared here 
 rb_data_object_alloc(VALUE klass, void *data, RUBY_DATA_FUNC dmark, RUBY_DATA_FUNC dfree)  
 ^~~~~~~~~~~~~~~~~~~~ 
compiling rbgtktextattributes.c 
compiling rbgtkbin.c  
compiling rbgtkscrolledwindow.c 
compiling rbgtkspinner.c  
compiling rbgtkbuilder.c  
compiling rbgtktooltip.c  
compiling rbgtkbbox.c 
compiling rbgtkhbox.c 
compiling rbgtkgamma.c
compiling rbgtkentrycompletion.c
compiling rbgdkvisual.c
compiling rbgtkspinbutton.c
compiling rbgtkmenushell.c
compiling rbgtktreedragdest.c
compiling rbgtktreemodelfilter.c
compiling rbgtkicontheme.c
compiling rbgtktreeselection.c
compiling rbgtktargetlist.c
compiling rbgtktearoffmenuitem.c
compiling rbgtkuimanager.c
compiling rbgtklayout.c
compiling rbgtkcelleditable.c
compiling rbgtkprogressbar.c
compiling rbgtkaccelgroup.c
compiling rbgtkiconsource.c
compiling rbgtktextappearance.c
compiling rbgtkcolorbutton.c
compiling rbgtkprintunixdialog.c
compiling rbgtkexpander.c
compiling rbgtkrc.c
compiling rbgtktooltips.c
compiling rbgtkmenuitem.c
compiling rbgtkrecentchooserwidget.c
compiling rbgtkvscale.c
compiling rbgdkconst.c
compiling rbgtkprintsettings.c
In file included from /Programs/Ruby/2.5.0/lib/ruby/gems/2.5.0/gems/glib2-3.2.1/ext/glib2/rbgobject.h:28:0,
 from rbgtk.h:28,
 from global.h:27,
 from rbgtkprintsettings.c:22:
/Programs/Ruby/2.5.0/lib/ruby/gems/2.5.0/gems/glib2-3.2.1/ext/glib2/rbglib.h:27:10: fatal error: rbglibdeprecated.h: No such file or directory
 #include "rbglibdeprecated.h"
  ^~~~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:242: rbgtkprintsettings.o] Error 1

make failed, exit code 2

Gem files will remain installed in /Programs/Ruby/2.5.0/lib/ruby/gems/2.5.0/gems/gtk2-3.2.1 for inspection.
Results logged to /Programs/Ruby/2.5.0/lib/ruby/gems/2.5.0/extensions/x86_64-linux/2.5.0-static/gtk2-3.2.1/gem_make.out
```

The error appears to be that rbglibdeprecated.h can not be found.

I did a locate:

```console
% locate rbglibdeprecated.h
```

The .h file exists:

```text
    /Depot/Temp/rubygnome2-3.2.1/glib2/ext/glib2/rbglibdeprecated.h
    /Programs/Ruby/2.4.3/lib/ruby/site_ruby/2.4.0/x86_64-linux/rbglibdeprecated.h
    /Programs/Ruby/2.5.0/lib/ruby/gems/2.5.0/extensions/x86_64-linux/2.5.0-static/glib2-3.2.1/rbglibdeprecated.h
    /Programs/Ruby/2.5.0/lib/ruby/gems/2.5.0/gems/glib2-3.2.1/ext/glib2/rbglibdeprecated.h
    /Programs/Ruby/2.5.0/lib/ruby/gems/2.5.0/gems/glib2-3.2.1/lib/rbglibdeprecated.h
```

I will probably install gtk2 again but will use /usr/ as prefix - perhaps that error will disappear
then. Please feel free to close this issue at any moment in time.

**Comment by kou** [comment]
Can you show build command line for `rbgtkprintsettings.c`?
We can show it by `make V=1` but I don't know how to do it with `gem install`...

**Comment by shevegen** [comment]
Hello kou,

Thank you for your help, but I think it was a user error (my system was setup incorrectly).

I re-installed the system (actually using absolute-linux right now rather than slackware; the latter is a bit slow in release). I compiled ruby 2.5.0 from source and then just did "gem install gtk2" - and it works very well. :)

I think the problem I had earlier was somewhat related that I did not compile gtk etc... into /usr/ prefix but somewhere else; and also had multiple several versions of gtk, glib, atk etc...

I am now using just one version and the oldschool /usr/ prefix and it works fine.

I will close this now since it should not be relevant anymore (because the error has been resolved, gem install works fine). By the way, I will from now on use gem install gtk2/gtk3 rather than the old setup.rb way - I believe in the long run, this is also easier for everyone.

The gem install way works very well! I have not yet tried gtk3 since I am still stuck mostly on gtk2 but eventually I will test with gtk3 too (I am very slow).

Thanks again for the help - closing the issue now.
